tests/chebyshev_test.py

```diff
@@ -0,0 +1,795 @@
+import numpy as np
+from numpy.polynomial import chebyshev as ncs
+import numpy.polynomial.polyutils as npu
+from numpy.polynomial.polyutils import RankWarning
+
+from absl.testing import absltest
+from absl.testing import parameterized
+
+from jax import numpy as jnp
+from jax._src import test_util as jtu
+import jax.numpy.polynomials as pu
+from jax.numpy.polynomials import chebyshev as cs
+
+from jax.config import config
+config.parse_flags_with_absl()
+
+if config.jax_enable_x64:
+  all_dtypes = ["int64", "float64", "complex128"]
+else:
+  all_dtypes = ["int32", "float32", "complex64"]
```

Comment:
Better would be to use `jtu.dtypes`, which contains the enable_x64 logic

Reply:
Using all_dtypes = jtu.dtypes.floating + jtu.dtypes.integer + jtu.dtypes.complex, each test runs with int32, int64, float32, float64, complex64, and complex128 dtypes when JAX_ENABLE_x64=true. Some of the tests fail when using float32 due to value mismatches. 

Removing jit decorators and replacing any compiled sections inside the function seems to fix this problem. For example, chebval fails one float32 test case and chebgrid2d (which calls chebval) fails two float32 test cases when jit decorators and JAX control flow are used. Replacing/removing these allows these tests to succeed. 

Doesn't work:
```
@partial(jit, static_argnames=('tensor',))
def chebval(x, c, tensor=True):
  [irrelevant code here]
  def body_fun(i, val):
    return (c[-i] - val[1], val[0] + val[1]*x2)

c0, c1 = lax.fori_loop(5, len(c)+1, body_fun, (c0, c1))
```

Still doesn't work:
```
@partial(jit, static_argnames=('tensor',))
def chebval(x, c, tensor=True):
  [irrelevant code here]
  for i in range(5, len(c) + 1):
    c0, c1 = (c[-i] - c1, c0 + c1 * x2)
```
Works for chebval, but chebgrid2d float32 test cases still fail:
```
def chebval(x, c, tensor=True):
  [irrelevant code here]
  def body_fun(i, val):
    return (c[-i] - val[1], val[0] + val[1]*x2)

c0, c1 = lax.fori_loop(5, len(c)+1, body_fun, (c0, c1))
```


Does work:
```
def chebval(x, c, tensor=True):
  [irrelevant code here]
  for i in range(5, len(c) + 1):
    c0, c1 = (c[-i] - c1, c0 + c1 * x2)
```

Any thoughts or suggestions for how to fix this while keeping JAX control flow and compilation? This issue only happens for a few float32 tests, and the relative difference from NumPy values is on the order of 1e-5.